subsys/net/buf.c

```diff
@@ -92,17 +92,18 @@ static u8_t *generic_data_ref(struct net_buf *buf, u8_t *data)
 	return data;
 }
 
-static u8_t *mem_pool_data_alloc(struct net_buf *buf, size_t size,
+static u8_t *mem_pool_data_alloc(struct net_buf *buf, size_t *size,
 				 s32_t timeout)
 {
 	struct net_buf_pool *buf_pool = net_buf_pool_get(buf->pool_id);
 	struct k_mem_pool *pool = buf_pool->alloc->alloc_data;
 	struct k_mem_block block;
 	u8_t *ref_count;
 
-	/* Reserve space for ref-count (u8_t) */
-	size += sizeof(struct k_mem_block_id) + 1;
-	if (k_mem_pool_alloc(pool, &block, size, timeout)) {
+	/* Reserve extra space for k_mem_block_id and ref-count (u8_t) */
+	if (k_mem_pool_alloc(pool, &block,
+			     sizeof(struct k_mem_block_id) + 1 + *size,
+			     timeout)) {
 		return NULL;
 	}
 
```

Comment:
Should assign to *size?

Reply:
No, since this size is what gets set to buf->size, i.e. it should not take into account internal metadata such as the refcount.